### ISSUE TITLE
Reproduction Gap: Official checkpoint (YuankaiLuo/SimVLA-LIBERO) does not reproduce paper numbers, especially on LIBERO-Long

### ISSUE THREAD

**Issue by cykbb** [LUOyk1999/SimVLA#7]
## Environment
- Checkpoint: `YuankaiLuo/SimVLA-LIBERO` (ckpt-150000, downloaded from HuggingFace)
- transformers: 4.57.6
- Evaluation: official `serve_smolvlm_libero.py` + `libero_client.py`, 10 trials/task
- Loading method: manual `load_state_dict()` with `safetensors` (no meta-device issue,
  no missing/unexpected key warnings in server log)
- norm_stats: `libero_norm.json` loaded correctly
- Image normalization: ImageNet (0.485/0.456/0.406), 384×384 — consistent with training

## Results with Official Checkpoint

| Task Suite     | My Results  | Paper (claimed) |
|----------------|-------------|-----------------|
| LIBERO-Spatial | 73%         | ?               |
| LIBERO-Object  | 60%         | ?               |
| LIBERO-Goal    | 68%         | ?               |
| LIBERO-Long    | **9%**      | ?               |

I also tested checkpoints trained locally following your training scripts
(`simvla_libero_large`, `simvla_libero_small`) — all show similarly low performance
on LIBERO-Long (3–9%), and much worse on LIBERO-Object (6–36%).

## What I Checked

I ruled out the following common issues:

1. **Weight loading**: Server log shows no missing/unexpected keys. I use manual
   `load_state_dict()`, not `from_pretrained` with meta-device context (see Issue #2).
2. **Image normalization**: Both training (`dataset_smolvlm.py`) and inference
   (`serve_smolvlm_libero.py`) use ImageNet normalization. The processor even has
   a comment warning about the SigLIP vs ImageNet mismatch.
3. **Image resolution**: 128×128 LIBERO images resized to 384×384, consistent with
   Issue #5 clarification.
4. **norm_stats**: Loaded correctly, state (8D) and action (7D) stats confirmed in log.
5. **Max steps**: libero_10 uses 900 steps, failures report steps=910 (900 + 10 wait
   steps) — working as expected.

## Question

**Could you clarify which exact checkpoint was used to generate the paper's reported
numbers?**

The 9% on LIBERO-Long is consistent across all checkpoints I tested (both the
official HuggingFace weight and locally trained ones), which suggests this may be
a fundamental limitation of the released model rather than an evaluation bug on
my side.

Specifically:
- Is `YuankaiLuo/SimVLA-LIBERO` the same checkpoint used in the paper, or a different
  (possibly smaller/earlier) version?
- Were the paper results generated with a different training configuration,
  larger batch size, more data, or a different model variant?
- Is there a better checkpoint available that achieves the numbers reported in the paper?

Thank you for the great work on this project. I just want to make sure I am
evaluating the right model.

**Comment by LUOyk1999** [comment]
Hi @cykbb, thanks for the detailed report!
The most likely root cause is using load_state_dict() instead of from_pretrained(), which causes critical model configuration to be lost.
Why this matters
SmolVLMVLA.from_pretrained(checkpoint_path) loads both the model weights and the config.json, which contains essential fields. When you use manual load_state_dict(), only the weights are loaded. 

Use from_pretrained() to load the model, not load_state_dict():
```
   from models.modeling_smolvlm_vla import SmolVLMVLA
   model = SmolVLMVLA.from_pretrained(checkpoint_path)
```

**Comment by FlamieZhu** [comment]
Hi @LUOyk1999, thanks for your outstanding work!

While training with the official scripts, I also noticed a reproduction gap. I used DeepSpeed on 4× A6000 GPUs and followed the settings in [train_smolvlm_large.sh.](https://github.com/LUOyk1999/SimVLA/blob/main/train_smolvlm_large.sh) The success rate on LIBERO-10 is around 88% (150k steps), which is below the 96% reported in the paper.

Could there be any missing details or subtle configuration differences that might explain this discrepancy? Any guidance would be greatly appreciated.

Thanks again!
